akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -576,7 +576,7 @@ object Uri {
     }
     def toMultiMap: Map[String, List[String]] = {
       @tailrec def append(map: Map[String, List[String]], q: Query): Map[String, List[String]] =
-        if (q.isEmpty) map else append(map.updated(q.key, q.value :: map.getOrElse(q.key, Nil)), q.tail)
+        if (q.isEmpty) map else append(map.updated(q.key, map.getOrElse(q.key, Nil) :+ q.value), q.tail)
```

Comment:
Yep, I was worried about the performance cost too.

Decide to submit the PR anyway because I guess the additional cost will be relevant only if we have a large number of parameters with the same key, which is not usual.